code-challenges.test.js

```diff
@@ -36,8 +67,54 @@ const fullArr2 = ["hello", 7, 23, -823, false, 78, null, "67", 6, "number"]
 // Expected output: [-823, 7, 23]
 
 
+describe ("oddNums", () => {
+  const fullArr1 = [4, 9, 0, "7", 8, true, "hey", 7, 199, -9, false, "hola"]
+  const fullArr2 = ["hello", 7, 23, -823, false, 78, null, "67", 6, "number"]
+  it ("returns a new array of only odd numbers from least to greatest", () => {
+  expect (oddNums(fullArr1)).toEqual([-9, 7, 9, 199])
+  expect (oddNums(fullArr2)).toEqual([-823, 7, 23])
+  })
+
+})
+//ReferenceError: oddNums is not defined
+
 // b) Create the function that makes the test pass.
 
+//psuedo code:
+//declare a fucntion oddNums
+//declare a new array 
+//set conditions for new array
+//use .filter to go through the array
+
+//1st try// needed something to be able to read the array beause I have more than just numbers in it. 
+// const oddNums = (array) => {
+//   let newArr = []
+//   for(let i=0; i<array.length; i++){
+//     if(array[i] % 2 === 1){
+//       newArr.push(array[i])
+//     }
+//   }
+//   return newArr
+// }
+////////////////
+//2nd try
+
+const oddNums = (array) => {
+  return array.filter(value => {
+      if(value % 2 !== 0 && typeof (value) === "number"){
+          return array
+      }
+  }).sort((a, b) => a - b)
+}
```

Comment:
This looks great!  I appreciate you leaving in your first attempt as well so I can see your process.

Reply:
yay!!